ui/src/Message.elm

```diff
@@ -101,33 +109,21 @@ update msg model token =
             in
             ( { model | channels = updatedChannel }, Cmd.map ChannelMsg channelCmd )
 
-        SelectedThing thingid ->
+        SelectedThing thingid thingkey ->
             let
                 ( updatedChannel, channelCmd ) =
                     Channel.update (Channel.RetrieveChannelsForThing thingid) model.channels token
             in
-            ( { model | thingid = thingid, channels = updatedChannel }, Cmd.map ChannelMsg channelCmd )
+            ( { model | thingid = thingid, channels = updatedChannel, token = thingkey }, Cmd.map ChannelMsg channelCmd )
```

Comment:
Stick to the terminology, i.e. be consistent. If you use `thingid = thingid` in the model, please use also `token = token` or, better `thingkey = thingkey`. Change the model to hold `thingkey` instead of `token` (`token` is ambivalent and can refer to user token).

Reply:
I agree. 
Mainflux docs itself is not consistent in naming this key/token, that's why it's mixed up here. It's referred to as `thing_key` and `thing_token` here: https://mainflux.readthedocs.io/en/latest/messaging/